### DISCUSSION TITLE
Error: Atom expected

### DISCUSSION THREAD

**Discussion by smccauliff** [ocaml/dune#5744] [General]
I'm new to ocaml and dune so this is probably a bad question.  Sorry about that.

I'm trying to build multiple executables.  I keep getting something that looks like a syntax error.  Then I tried to run an example from the dune documentation and it gets a similar error. "opam upgrade dune" says everything is up to date

This is from the dune documentation:
(executables
 ((names (a b c))
  (modes ((byte exe) (native shared_object)))))

File "dune", line 2, characters 2-17:
2 |  ((names (a b c))
      ^^^^^^^^^^^^^^^
Error: Atom expected

My goal is just to write several example programs to learn ocaml.  All the .ml files are in a single directory.  If reorganizing things would help then I could do that, but it's not clear how to do that either.

Thanks

**Comment by ghost** [comment]
The error message is not great and could definitely be improved. You need to remove a few parentheses:

```scheme
(executables
 (names (a b c)
 (modes (byte exe) (native shared_object)))
```

Could you point to where this example comes from the doc? Or better, submit a PR :)

**Comment by smccauliff** [comment]
When this is paste into the dune file I get:
dune build
File "dune", line 4, characters 1-1:
4 |  
     
Error: unclosed parenthesis at end of input

Adding an additional parenthesis (names (a b c))
2 |  (names (a b c))
            ^^^^^^^
Error: Atom or quoted string expected

Adding the ) at the end:
File "dune", line 2, characters 8-15:
2 |  (names (a b c)
            ^^^^^^^
Error: Atom or quoted string expected

**Comment by nojb** [comment]
Try removing the parentheses before "a" :)
```
(executables
 (names a b c)
 (modes (byte exe) (native shared_object)))
```
